private/bufpkg/bufmodule/bufmoduleapi/uploader.go

```diff
@@ -140,30 +141,6 @@ func (a *uploader) Upload(
 	}
 
 	var v1beta1ProtoUploadRequestContents []*modulev1beta1.UploadRequest_Content
-	if len(uploadOptions.Labels()) > 0 {
```

Comment:
Moved this last (and refactored to use `else` since Labels/BranchOrDraft/Tags are mutually exclusive.